Advanced Memory Dashboards

azat opened the pull request:
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Advanced Memory Dashboards

![image](https://github.com/user-attachments/assets/6eed2471-5427-4b18-9cae-272a338eb5b2)


_Note: first commit contains cleanup of dashboard queries_

azat committed:
Prettify all dashboard queries

azat committed:
Advanced Memory Dashboards

azat committed:
Add a test for system.dashboards (just a smoke test that all queries works)

azat committed:
Do not check curly braces in StorageSystemDashboards.cpp

azat committed:
Fix in-memory chart on dashboards

antaljanosbenjamin commented on ci/jobs/scripts/check_style/check_cpp.sh:
```diff
@@ -33,6 +33,7 @@ function in_array()
 
 find $ROOT_PATH/{src,base,programs,utils} -name '*.h' -or -name '*.cpp' 2>/dev/null |
     grep -vP $EXCLUDE |
+    grep -v 'src/Storages/System/StorageSystemDashboards.cpp' |
```
THe curly braces are problematic because of the new formatting, I guess, right?

azat replied:
Yes

azat committed:
tests: do not read ci-logs cluster in 03540_system_dashboards

azat commented:
>[PR / Stateless tests (asan, flaky check) (pull_request)](https://github.com/ClickHouse/ClickHouse/actions/runs/15570960287/job/43854788023?pr=81589)Failing after 17m

Seems that the runner got KILLed

azat committed:
tests: fix 03540_system_dashboards under ASan (use curl over client binary)

azat committed:
tests: more stable 03540_system_dashboards

azat commented:
>[Build (amd_debug)](https://s3.amazonaws.com/clickhouse-test-reports/json.html?PR=81589&sha=0018ae04f8fd8066d4a2ced3bd1ef958b7a0e090&name_0=PR&name_1=Build%20%28amd_debug%29) — Job got terminated with an error, exit code [-15]

@maxknv any clue?

azat committed:
tests: 03540_system_dashboards is still flaky in flaky check, forbid it

azat committed:
tests: disable distributed plan for 03540_system_dashboards

CI: https://s3.amazonaws.com/clickhouse-test-reports/json.html?PR=81589&sha=0018ae04f8fd8066d4a2ced3bd1ef958b7a0e090&name_0=PR&name_1=Stateless%20tests%20%28asan%2C%20distributed%20plan%2C%201%2F2%29
